src/lib.rs

```diff
@@ -163,7 +163,7 @@ impl<R: Read> Iterator for MessageStream<R> {
                         return None;
                     } else {
                         self.in_error_state = true;
-                        return Some(Err(format!("Parse failed: {:?}", e.into_error_kind()).into()));
+                        return Some(Err(format!("Parse failed: {:?}, buffer context {:?}", e.into_error_kind(), &self.buffer[self.bufstart..self.bufstart + 20]).into()));
```

Comment:
I'm happy to remove this - I found it useful to get some data context when hitting an error.